omni-relayer/src/workers/eth.rs

```diff
@@ -0,0 +1,83 @@
+use std::sync::Arc;
+
+use anyhow::Result;
+use futures::future::join_all;
+
+use alloy::rpc::types::Log;
+use ethereum_types::H256;
+use nep141_connector::Nep141Connector;
+
+use crate::{config, utils};
+
+pub async fn finalize_withdraw(
+    config: config::Config,
+    redis_client: redis::Client,
+    connector: Arc<Nep141Connector>,
+) -> Result<()> {
+    let redis_connection = redis_client.get_multiplexed_tokio_connection().await?;
+
+    loop {
+        let mut redis_connection_clone = redis_connection.clone();
+        let Some(mut events) = utils::redis::get_events(
+            &mut redis_connection_clone,
+            config.redis.near_sign_transfer_events.clone(),
+        )
+        .await
+        else {
+            tokio::time::sleep(tokio::time::Duration::from_secs(
+                config.redis.sleep_time_after_events_process_secs,
+            ))
+            .await;
+            continue;
+        };
+
+        let mut handlers = Vec::new();
+        while let Some((key, event)) = events.next_item().await {
+            if let Ok(withdraw_log) =
+                serde_json::from_str::<Log<crate::startup::eth::Withdraw>>(&event)
+            {
+                handlers.push(tokio::spawn({
+                    let config = config.clone();
+                    let mut redis_connection = redis_connection.clone();
+                    let connector = connector.clone();
+
+                    async move {
+                        log::info!("Decoded log: {:?}", withdraw_log);
```

Comment:
We also should check fee here.

Reply:
I believe that we only need to check/claim fee during `NEAR -> X` transfers, not vice versa. At least that's what I remember from our discussion with @karim-en